Documentation/interacting_v3.md

```diff
@@ -0,0 +1,183 @@
+# Interacting with etcd
+
+Users mostly interact with etcd by putting or getting the value of a key. This section describes how to do that by using etcdctl, a command line tool for interacting with etcd server. The concepts described here should apply to the gRPC APIs or client library APIs.
+
+By default, etcdctl talks to the etcd server with the v2 API for backward compatibility. For etcdctl to speak to etcd using the v3 API, the API version must be set to version 3 via the `ETCDCTL_API` environment variable.
+
+``` bash
+export ETCDCTL_API=3
+```
+
+## Write a key
+
+Applications store keys into the etcd cluster by writing to keys. Every stored key is replicated to all etcd cluster members through raft to achieve consistency and reliability.
+
+Here is the command to set the value of key `foo` to `bar`:
+
+``` bash
+$ etcdctl put foo bar
+OK
+```
+
+## Read keys
+
+Applications can read values of keys from etcd cluster. Both ranged queries and single-key queries are supported. 
+
+Suppose we have keys already stored in etcd as blow:
+
+```
+foo = bar
+foo1 = bar1
+foo3 = bar3
+```
+
+Here is the command to read the value of key `foo`:
+
+```bash
+$ etcdctl get foo
+foo
+bar
+```
+
+Here is the command to range over the keys from `foo` to `foo9`:
+
+```bash
+$ etcdctl get foo foo9
+foo
+bar
+foo1
+bar1
+foo3
+bar3
+```
+
+## Read past version of keys
+
+Applications may want to read superseded versions of a key for various reasons. For example, an application may wish to rollback to an old configuration by accessing an early version of a key. Alternatively, an application may want a consistent view over multiple keys over multiple requests by accessing key history.
+Since every modification to the etcd cluster key-value store increments the global revision of etcd cluster, an application can read superseded keys by providing a historical etcd revision.
+
+Suppose an etcd cluster already has the following keys:
+
+``` bash
+$ etcdctl put foo bar         # revision = 2
+$ etcdctl put foo1 bar1       # revision = 3
+$ etcdctl put foo bar_new     # revision = 4 
+$ etcdctl put foo1 bar1_new   # revision = 5
+```
+
+Here are an example to access the past versions of keys:
+
+```bash
+$ etcdctl get foo foo10 # access the most recent versions of keys
+foo
+bar_new
+foo1
+bar1_new
+
+$ etcdctl get --rev=4 foo foo10 # access the versions of keys at revision 4
+foo
+bar_new
+foo1
+bar1
+
+$ etcdctl get --rev=3 foo foo10 # access the versions of keys at revision 3
+foo
+bar
+foo1
+bar1
+
+$ etcdctl get --rev=2 foo foo10 # access the versions of keys at revision 2
+foo
+bar
+
+$ etcdctl get --rev=1 foo foo10 # access the versions of keys at revision 1
+```
+
+## Delete keys
+
+Applications can delete a key or a range of keys from etcd cluster.
+
+Here is the command to delete key `foo`:
+
+```bash
+$ etcdctl del foo
+1 # one key is deleted
+```
+
+Here is the command to delete keys ranging from `foo` to `foo10`:
```

Comment:
It is just for consistency across the examples - the first range example used `foo` through `foo9`.